src/nekontrol/interactive/submit.py

```diff
@@ -0,0 +1,233 @@
+#
+# This file's contents is heavily inspired by
+# https://github.com/Kattis/kattis-cli, which is licensed under the MIT
+# license. Their copyright notice is included here.
+#
+# KATTIS LICENSE BEGINS
+#
+#   The MIT License (MIT)
+#
+#   Copyright (c) 2006-2015 Kattis
+#
+#   Permission is hereby granted, free of charge, to any person obtaining a copy
+#   of this software and associated documentation files (the "Software"), to deal
+#   in the Software without restriction, including without limitation the rights
+#   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+#   copies of the Software, and to permit persons to whom the Software is
+#   furnished to do so, subject to the following conditions:
+#
+#   The above copyright notice and this permission notice shall be included in all
+#   copies or substantial portions of the Software.
+#
+#   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+#   SOFTWARE.
+#
+# KATTIS LICENSE ENDS
+
+import os.path as path
+import re
+import time
+from lxml.html import fragment_fromstring
+
+import click
+import requests
+
+from .. import language, util
+from . import test
+
+_HEADERS = {"User-Agent": "nekontrol/0.2.5"}
+
+
+def login(user: str, token: str):
+    r = requests.post(
+        "https://open.kattis.com/login",
+        data={
+            "user": user,
+            "token": token,
+            "script": "true",
+        },
+        headers=_HEADERS,
+    )
+    if r.status_code != 200:
+        raise click.ClickException(f"Failed to login ({r.status_code}): {r.text}")
+    return r.cookies
+
+
+def get_submission_id(response):
+    m = re.search(r"Submission ID: (\d+)", response)
+    if m is None:
+        print(response)
+        raise click.ClickException("Couldn't find submission ID in response.")
+    return m.group(1)
+
+
+def submit(file_path, problem, config):
+    file_name = path.basename(file_path)
+    file_base, extension = path.splitext(file_name)
+
+    c = util.cw(config.color)
+
+    if problem is None:
+        if config.verbose:
+            print(c(f"No problem name specified, guessing '{file_base}'", "yellow"))
+        problem = file_base
+
+    lang = language.get_lang(file_path, config)
+
+    if lang is None:
+        raise click.ClickException(
+            f"Language for file extension {extension} is not implemented."
+        )
+
+    # test before submitting
+    test.test(file_path, problem, config)
+
+    if not config.force:
+        ans = input("Submit? [Y/n] ")
+        if ans.lower() == "n":
+            return
```

Comment:
Maybe it would be preferable to use click prompts, since this is not the best prompt.